app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationWithMultiManagersTest.kt

```diff
@@ -0,0 +1,154 @@
+          circleAnnotation = circleAnnotationManager.create(
+            CircleAnnotationOptions()
+              .withCircleColor(Color.RED)
+              .withCircleRadius(5.0)
+              .withPoint(Point.fromLngLat(0.0, 0.0))
+          )
+
+          polygonAnnotationManager = mapView.annotations.createPolygonAnnotationManager(mapView)
+          polygonAnnotation = polygonAnnotationManager.create(
+            PolygonAnnotationOptions()
+              .withPoints(points)
+              .withData(JsonPrimitive("Foobar"))
+              .withFillColor(Color.RED)
+          )
+
+          polylineAnnotationManager = mapView.annotations.createPolylineAnnotationManager(mapView)
+          polylineAnnotation = polylineAnnotationManager.create(
+            PolylineAnnotationOptions()
+              .withPoints(polylinePoints)
+              .withLineColor(Color.RED)
+              .withLineWidth(5.0)
+          )
+
+          Thread {
+            for (i in 0..updateTimes) {
+              pointAnnotation.point = Point.fromLngLat(
+                pointAnnotation.point.longitude() + updateSteps,
+                pointAnnotation.point.latitude() + updateSteps
+              )
+              pointAnnotationManager.update(pointAnnotation)
+              Thread.sleep(10)
+            }
+            latch.countDown()
+          }.start()
+
+          Thread {
+            for (i in 0..updateTimes) {
+              circleAnnotation.point = Point.fromLngLat(
+                circleAnnotation.point.longitude() + updateSteps,
+                circleAnnotation.point.latitude() + updateSteps
+              )
+              circleAnnotationManager.update(circleAnnotation)
+              Thread.sleep(10)
+            }
+            latch.countDown()
+          }.start()
+
+          Thread {
+            for (i in 0..updateTimes) {
+              points[0][0] = Point.fromLngLat(
+                points[0][0].longitude() + updateSteps,
+                points[0][0].latitude() + updateSteps
+              )
+              polygonAnnotation.points = points
+              polygonAnnotationManager.update(polygonAnnotation)
+              Thread.sleep(10)
+            }
+            latch.countDown()
+          }.start()
+          Thread {
+            for (i in 0..updateTimes) {
+              polylinePoints[0] = Point.fromLngLat(
+                polylinePoints[0].longitude() + updateSteps,
+                polylinePoints[0].latitude() + updateSteps
+              )
+              polylineAnnotation.points = polylinePoints
+              polylineAnnotationManager.update(polylineAnnotation)
+              Thread.sleep(10)
+            }
+            latch.countDown()
+          }.start()
+        }
+      }
+    }
+    if (!latch.await(3000, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+    assertEquals(pointAnnotation.point.latitude(), updateSteps * updateTimes, 0.1)
```

Comment:
0.1 should be moved to constants and also I guess 0.1 is too big. Let's use smth like 0.0001